File: 1.two-sun/python/main.py
class Solution:
    def twoSum(self, nums, target):
        dict = {}

        for i in range(len(nums)):
            if target - nums[i] not in dict:
                dict[nums[i]] = i
            else:
                return [dict[target - nums[i]], i]

sol=Solution()
print(sol.twoSum([1,2,6],8))

File: 1.two-sun/java/Main.java
import java.util.HashMap;
public class Main {
    public static void main(String[] args) {
        int[] nums={5,7,9,11,13};
        int target = 12;
        int[] ans;
        ans=twoSum(nums, target);
        System.out.println(ans[0]+","+ans[1]);
    }

    public static int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer> map=new HashMap<>();
        for(int i=0;i<nums.length;i++){
            int sub=target-nums[i];
            if(map.containsKey(sub)){
                return new int[]{i,map.get(sub)};
            }
            map.put(nums[i], i);
        }
        throw new IllegalArgumentException("No two sum solution");
    }
    
    
}